### PR TITLE
Select: add the ability to set a prefix

### PR DESCRIPTION
**Pull request by vodkabears**
Like this one: https://yadi.sk/i/HL6bAea2s8QbD

### CODE REVIEW

**Comment by ruslankerimov** [comment]
I propose to make more flexible, e.g. using property `pattern = "Day: %value"` or `decorator = function(value) { return 'Day: ' + value; }`.

**Comment by vodkabears** [comment]
+

**Comment by narqo** [comment]
To be fair, I don't like both variants, as they solve only particular issue, but don't introduce any API points for further extends. For example, `select` from bem-components allows to customise it in a way below:

screen shot 2016-05-31 at 13 36 18

or even this way

screen shot 2016-05-31 at 13 31 04

Currently, I suggest to introduce protected method `Select#getButtonText()` and to override it on the project level in the modified component. E.g. 

``` js
class SelectWithLabel extends Select {
    /** @verride **/
    getButtonText() {
        return `${this.props.label}: ${super.getButtonText()}`;
    }
}
```

/cc @pasaran @doochik

**Comment by pasaran** [comment]
Probably it's better to have an array of selected items, not just joined text.

**Comment by vodkabears** [comment]
Ok, how about this solution:
https://gist.github.com/VodkaBears/8461cf247a096440cb7fd0a3c18cb9ba.

> Currently, I suggest to introduce protected method Select#getButtonText() and to override it on the project level in the modified component. E.g.

Anyway, the package has an unstable version, so it is ok if the public API will change in 0.x minor versions, but undocumentated methods may change at any time.

**Comment by vodkabears** [comment]
Updated.

> Anyway, the package has an unstable version, so it is ok if the public API will change in 0.x minor versions, but undocumentated methods may change at any time.
> 
> Currently, I suggest to introduce protected method Select#getButtonText() and to override it on the project level in the modified component. E.g.

For the history: we have decided to make a protected method which will be the part of the API.

File: src/Select/index.js [hunk 1]
```diff
@@ -184,6 +172,22 @@ class Select extends Component {
         return this.refs.menu;
     }
 
+    getButtonText() {
```
Comment (vodkabears) [inline src/Select/index.js]: maybe `renderButtonText` or `renderButtonContent`?

File: src/Select/index.js [hunk 2]
```diff
@@ -184,6 +172,22 @@ class Select extends Component {
...
+                if (value.length === 1) {
+                    res.push(Component.textValue(item));
+                } else {
+                    res.push(item.props.checkedText || Component.textValue(item));
+                }
+            }
+            return res;
+        }, []);
+
+        return text.join(', ') || this.props.placeholder || '—';
```
Comment (vodkabears) [inline src/Select/index.js]: Another question: should this logic be in the method?

File: src/Select/index.js (hunk 1 above)
Comment (narqo) [inline src/Select/index.js]: `renderButtonText` sounds nice to me

File: src/Select/index.js (hunk 2 above)
Comment (narqo) [inline src/Select/index.js]: If we want this method to be overridable, I think there is nothing particular wrong for it to be here.

Comment (vodkabears) [inline src/Select/index.js]: Yes, but it is easier to override when the base method doesn't know about placeholders.

```
renderButtonText() {
  const text = super.renderButtonText();

  return text && `Days: ${text}`;
}
```

Conceptually, seems like it's ok to show placeholders if `renderButtonText` returns nothing.

File: src/Select/index.js [hunk 3]
```diff
@@ -97,7 +101,7 @@ class Select extends Component {
                 onClick={this.onButtonClick}
                 onKeyDown={this.onButtonKeyDown}
             >
-                {buttonText}
+                {this.renderButtonText() || this.props.placeholder || '—'}
```
Comment (vodkabears) [inline src/Select/index.js]: Technically, '-' should be the default value of the placeholder.

**Comment by vodkabears** [comment]
up

**Comment by pasaran** [comment]
Looks ok.
But I would use `renderSomething` for jsx-methods (returning `React.Element`).
I think `getButtonText` is more appropriate. But it's not that important, `renderButtonText` ok too.

**Comment by narqo** [comment]
Nice!

**Comment by narqo** [comment]
I'll publish a new release latter this day

**Pull request merged**